From eff17cb90e74149b1536ba5e2acfa6b5c2c6619e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sun, 2 Feb 2025 17:51:46 +0000
Subject: [PATCH] Check path if exist

This fixes the problem that no fallback happens to the English help files when localized variants are missing.

Fixes 7df337f5bf980f004dd9d10fb91e5052f35d9e46
---
 .../src/openpaperwork_core/resources/setuptools.py       | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/openpaperwork-core/src/openpaperwork_core/resources/setuptools.py b/openpaperwork-core/src/openpaperwork_core/resources/setuptools.py
index 18585a1df..efc8a0a98 100644
--- a/openpaperwork-core/src/openpaperwork_core/resources/setuptools.py
+++ b/openpaperwork-core/src/openpaperwork_core/resources/setuptools.py
@@ -1,4 +1,5 @@
 import logging
+import os
 try:
     import importlib.resources
     IMPORTLIB_RESOURCES_AVAILABLE = True
@@ -29,6 +30,14 @@ class Plugin(PluginBase):
             return
 
         path = importlib.resources.files(pkg).joinpath(filename)
+
+        if not os.access(path, os.R_OK):
+            LOGGER.warning(
+                "Failed to find resource file %s/%s,"
+                " tried at path %s.", pkg, filename, path
+            )
+            return None
+
         LOGGER.debug("%s:%s --> %s", pkg, filename, path)
         return importlib.resources.as_file(path)
 
-- 
GitLab

